sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/hash.scala

```diff
@@ -282,6 +282,7 @@ abstract class HashExpression[E] extends Expression {
     }
 
     val hashResultType = CodeGenerator.javaType(dataType)
+    val typedSeed = if (dataType.sameType(LongType)) s"${seed}L" else s"$seed"
```

Comment:
Thank you for making a PR, @patrickcording . BTW, this seems to change the hash result, doesn't it?

Reply:
Would it change the result? it would just let it not fail.

Reply:
It doesn't change the result. For xxHash, the generated code just becomes `long varName = 123L` instead of `long varName = 123`. When the seed is <= 2^31, theres no difference between the two statements. When it is > 2^31, compilation of the generated code would fail without the `L`, so it is now possible to use any 64 bit seed.

Reply:
Yeah I think the concern was something like: a 4-byte int seed isn't the same as an 8-byte long seed when it comes to hashing, even if they have the same integer value. But here, this should only affect `HashExpression[Long]`s like `XxHash64`, which already would have promoted any int seed value to long in generated code.